src/js_ast.zig

```diff
@@ -2539,33 +2529,31 @@ pub const E = struct {
             }
         }
 
-        pub fn toJS(s: *String, allocator: std.mem.Allocator, globalObject: *JSC.JSGlobalObject, comptime opts: ToJsOpts) JSC.JSValue {
+        pub fn toJS(s: *String, allocator: std.mem.Allocator, globalObject: *JSC.JSGlobalObject, comptime _: ToJsOpts) !JSC.JSValue {
+            s.resolveRopeIfNeeded(allocator);
             if (!s.isPresent()) {
                 var emp = bun.String.empty;
                 return emp.toJS(globalObject);
             }
 
-            if (s.is_utf16) {
-                var out, const chars = bun.String.createUninitialized(.utf16, s.len());
+            if (s.isUTF8()) {
+                if (try strings.toUTF16Alloc(allocator, s.slice8(), false, false)) |utf16| {
```

Comment:
We might need a test with invalid utf8

Reply:
Actually, should this replace invalid sequences?

Reply:
`"abc"[0]` constant folding expects non-utf16 E.Strings to be ascii-only, and there is probably more code that does also. I think it would make sense to enforce E.String be either ascii or utf-16 rather than utf-8 or utf-16

Reply:
I agree. That makes sense